### PR TITLE
Sam/Caleb submission

### PR DESCRIPTION
**Pull request by calebji0**

### CODE REVIEW

File: src/main/java/com/statefarm/codingcomp/agent/AgentLocator.java [hunk 1]
```diff
@@ -42,19 +72,59 @@
...
+		for(Agent a: all){
+			if(nameN.containsKey(a.getName())){
+				nameN.put(a.getName(), nameN.get(a.getName())+1);
+			}else{
+				nameN.put(a.getName(), 1);
+			}
+		}
+		
+		// add exception if it's empty?
+		String mostPopular = nameN.keySet().iterator().next();
```
Comment (aaomidi) [inline src/main/java/com/statefarm/codingcomp/agent/AgentLocator.java:111]: You do realize hashmaps don't have order, right?

File: src/main/java/com/statefarm/codingcomp/agent/AgentLocator.java [hunk 2]
```diff
@@ -42,19 +61,65 @@
...
+		return names;
 	}
 
 	public String mostPopularFirstName() {
-		return null;
+		
+		HashMap<String, Integer> nameN = new HashMap<String, Integer>();
+		
+		List<Agent> all = getAllAgents();
+		// Where do the names come from?
```
Comment (Vilsol) [inline src/main/java/com/statefarm/codingcomp/agent/AgentLocator.java:101]: You know when mommy and daddy spend a lot of time together... yeah, you get the rest...

File: src/main/java/com/statefarm/codingcomp/agent/AgentLocator.java [hunk 3]
```diff
@@ -42,19 +61,65 @@
...
+		for (String fileName: sfFileReader.findAgentFiles()){
+			allAgents.add(agentParser.parseAgent(fileName));	
+		}
+		return allAgents;
+	} 
 
 	public Map<String, List<Agent>> getAllAgentsByUniqueFullName() {
-		return null;
+		
+		HashMap names = new HashMap<String, List<Agent>>();
```
Comment (zackpollard) [inline src/main/java/com/statefarm/codingcomp/agent/AgentLocator.java:85]: If you did `Map<String, List<Agent>> names = new HashMap<>();` then you wouldn't need to cast to ArrayList later

File: src/main/java/com/statefarm/codingcomp/agent/AgentLocator.java (hunk 1 above)
Comment (calebji0) [reply src/main/java/com/statefarm/codingcomp/agent/AgentLocator.java:111]: but their keysets do?

Comment (zackpollard) [reply src/main/java/com/statefarm/codingcomp/agent/AgentLocator.java:111]: No, the keyset is generated from the HashMap, a hashmap can be in any order and the keyset is generated when you call keySet(). Also, the Set is not a TreeSet and therefore can also shuffle into any order

**Pull request closed without merging**